packages/keystone/preprocessors/customAccess.js

```diff
@@ -0,0 +1,151 @@
+const { parseListAccess, parseFieldAccess } = require('@keystonejs/access-control')
+const cloneDeep = require('lodash/cloneDeep')
+const get = require('lodash/get')
+const isEmpty = require('lodash/isEmpty')
+
+const { GQL_LIST_SCHEMA_TYPE } = require('@open-condo/keystone/schema')
+
+const DEFAULT_PERMISSIONS = { read: false, update: false, create: false, delete: false }
+const DEFAULT_FIELD_PERMISSIONS = { read: true, update: true, create: true }
+
+/**
+ *
+ * @param user {Object} keystone user
+ * @return {boolean}
+ */
+function hasDefinedPermissions (user) {
+    return get(user, 'customAccess', null) !== null
+}
+
+/**
+ * If there are some defined permissions for current user -> use it, otherwise return original one
+ * @param user {Object} keystone user
+ * @param permission {boolean} custom permission found for that user. False by default
+ * @param originalAccessResult {boolean} access based on code of current schema
+ * @return {*|boolean}
+ */
+function getResultPermission (user, permission, originalAccessResult) {
+    if (hasDefinedPermissions(user)) {
+        return permission === true || originalAccessResult
+    }
+
+    return originalAccessResult
+}
+
+function getListPermissions (user, listKey) {
+    return get(user, ['customAccess', 'lists', listKey, 'access'], DEFAULT_PERMISSIONS)
+}
+
+function getFieldPermissions (user, listKey, field) {
+    return get(user, ['customAccess', 'lists', listKey, 'fields', field], DEFAULT_FIELD_PERMISSIONS)
+}
+
+/**
+ * Creates a "dynamic" access style function, that's check for custom access for current field or list.
+ * If there is a customAccess provided for current user -> calculate result based on provided rules.
+ * Otherwise, return original access, that placed at the code
+ * @param permissionsGetter {Function} custom access field parser function
+ * @param originalAccessFn {Function} original access function defined by code
+ * @return {(function({authentication: {item: *}, operation: *}): Promise<*>)|*} keystone function style access
+ */
+function createDynamicAccessWrapper (permissionsGetter, originalAccessFn){
+    return async function customAsyncAccessWrapper ({ authentication: { item: user }, operation }) {
+        const permissions = permissionsGetter(...arguments)
+        arguments[0].permissions = permissions
+
+        const originalAccessResult = await originalAccessFn(...arguments)
+
+        return getResultPermission(user, permissions[operation], originalAccessResult)
```

Comment:
`permissions`, `permissions`, `permissions` makes ripples in my eyes ;)
Maybe this will be simpler

```suggestion
    return async function customAsyncAccessWrapper ({ authentication: { item: user }, operation, permissions }) {
        const modifiedPermissions = permissionsGetter(...arguments)
        const originalAccessResult = await originalAccessFn({ ...arguments, permissions: modifiedPermissions })

        return getResultPermission(user, modifiedPermissions[operation], originalAccessResult)
```